src/samecode/halohash.py

```diff
@@ -296,24 +297,19 @@ def build_buckets(self):
         Return a list of buckets splitting high and low using bit shifts and
         group by high.
         """
-        # FIXME use a defaultdict(list)
-        hash_buckets = {}
+        hash_buckets = defaultdict(list)
         for h in self.hashes:
             hbv = bitarray_from_bytes(h.digest())
             hi = bit_to_num(hbv[0:self.high])
             lo = hbv[self.high:]
-            try:
-                hash_buckets[hi].append(lo)
-            except KeyError:
-                hash_buckets[hi] = []
-                hash_buckets[hi].append(lo)
+            hash_buckets[hi].append(lo)
         buckets = []
         # TODO: are we off by one on this range?
         for i in range(self.number_of_buckets):
-            try:
-                buck = hash_buckets[i]
+             buck = hash_buckets[i]
+             if len(buck) > 0:
```

Comment:
Is this not changing the semantics? Would there be a way to have a unit test may be for this function?